### PR TITLE
[BZ-1269979] return a -1 for an SSLStream eof instead of 0

### PR DESCRIPTION
**Pull request by aogburn**
https://bugzilla.redhat.com/show_bug.cgi?id=1269979

### CODE REVIEW

**Comment by vtunka** [comment]
Fix not merged upstream:
http://hg.openjdk.java.net/jdk8u/jdk8u60/jdk/file/935758609767/src/share/classes/sun/net/httpserver/SSLStreams.java#l513

**Comment by darranl** [comment]
This is upstream - we have forked and are not tracking anything else now.

Also we stop using the HTTP Server as soon as Undertow was available anyway so normal upstream first does not apply here.

**Comment by wolfc** [comment]
Noted at https://github.com/jbossas/httpserver/wiki. I think we can proceed with merge and tag.

File: src/main/java/org/jboss/sun/net/httpserver/SSLStreams.java [hunk 1]
```diff
@@ -602,7 +602,7 @@ public int read (byte[] buf) throws IOException {
 
         public int read () throws IOException {
             int n = read (single, 0, 1);
-            if (n == 0) {
+            if (n == 0 || n == -1) {
```
Comment (darranl) [inline src/main/java/org/jboss/sun/net/httpserver/SSLStreams.java]: Could just be if (n <= 0) {

**Comment by istudens** [comment]
Merged

**Pull request merged**